frontend/src/views/CellGuide/components/common/OntologyDagView/components/AnimatedNodes/index.tsx

```diff
@@ -56,16 +60,24 @@ export default function AnimatedNodes({
   showTooltip,
   hideTooltip,
   cellTypesWithMarkerGeneStats,
+  setCellInfoCellType,
 }: AnimatedNodesProps) {
   const [timerId, setTimerId] = useState<NodeJS.Timer | null>(null); // For hover event
-
+  const router = useRouter();
   const handleAnimationEnd = (node: HierarchyPointNode<TreeNodeWithState>) => {
     // Update the starting position of the node to be its current position
     node.data.x0 = node.x;
     node.data.y0 = node.y;
   };
-
   const { data: cellTypeMetadata } = useCellTypeMetadata() || {};
+  const handleNodeLabelClick = ({ cellTypeId, cellTypeName }: CellType) => {
+    if (setCellInfoCellType) {
+      setCellInfoCellType({ cellTypeId, cellTypeName });
+    } else {
```

Comment:
if setCellInfoCellType is not defined, then just default to original label clicking behavior (used by TissueCards).